### PR TITLE
lib/client: Fix Error Handling

### PR DESCRIPTION
**Pull request by alrs**
This fixes two dropped errors in `lib/client` and does away with an `err` assignment that was being done for no reason.

### CODE REVIEW

File: lib/client/api.go [hunk 1]
```diff
@@ -1138,7 +1138,6 @@ func (tc *TeleportClient) Play(ctx context.Context, namespace, sessionID string)
 			return trace.Wrap(err)
 		}
 		if len(tmp) == 0 {
-			err = io.EOF
```
Comment (russjones) [inline lib/client/api.go]: You can also drop the `err == nil` in line 1135.

**Comment by webvictim** [comment]
ok to test

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Review by russjones — APPROVED** [review]

**Pull request merged**